### PR TITLE
add docstring into NearestNeighbors class

### PR DESCRIPTION
**Pull request by lijinf2**
Also the example codes for vector and multi-col input types

### CODE REVIEW

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/knn.py [hunk 1]
```diff
@@ -221,6 +247,31 @@ class NearestNeighbors(
...
+    ...         (2, Vectors.dense([3.0, 3.0]),)]
+    >>> data_df = spark.createDataFrame(data, ["id", "features"])
+    >>> query = [(3, Vectors.dense([1.0, 1.0]),),
+    ...          (4, Vectors.dense([3.0, 3.0]),)]
+    >>> query_df = spark.createDataFrame(query, ["id", "features"])
+    >>> topk = 2
+    >>> gpu_knn = NearestNeighbors().setInputCol("features").setIdCol("id").setK(topk)
+    >>> gpu_model = gpu_knn.fit(data_df)
+
+    >>> from spark_rapids_ml.knn import NearestNeighbors
```
Comment (leewyang) [inline python/src/spark_rapids_ml/knn.py:264]: I would remove this second import on the assumption that this is a single session.

Comment (leewyang) [reply python/src/spark_rapids_ml/knn.py:264]: And maybe add comment lines explaining the two varieties, e.g.
```
# single column input
...
# multi-column input
```

Comment (lijinf2) [reply python/src/spark_rapids_ml/knn.py:266]: did not delete the second import, thinking the example codes may be copied to a different session. 

array input is also single column so added the comments "vector column input" and "multi-column input".

**Comment by lijinf2** [comment]
build

**Review by leewyang — APPROVED** [review]

**Pull request merged**